Simcoe/Simcoe.swift

```diff
@@ -105,8 +105,9 @@ public final class Simcoe {
     /// - Parameters:
     ///   - product: The SimcoeProductConvertible instance.
     ///   - eventProperties: The event properties.
-    func logAddToCart<T: SimcoeProductConvertible>(_ product: T, eventProperties: Properties?) {
-        let providers: [CartLogging] = findProviders()
+    func logAddToCart<T: SimcoeProductConvertible>(_ product: T,
+                      eventProperties: Properties?, providers: [AnalyticsTracking] = Simcoe.engine.providers) {
```

Comment:
Looks good. You might want to note down somewhere in the documentation that if you don't specify `providers`, we will use the providers associated with the current instance of `Simcoe`.